datastream-server-restli/src/main/java/com/linkedin/datastream/server/dms/DatastreamStore.java

```diff
@@ -21,6 +21,16 @@
    */
   Datastream getDatastream(String key);
 
+  /**
+   * Retrieves raw data in json from the given path.
+   */
+  String getRawData(String path);
+
+  /**
+   * Retrieves connector task path specific to the store implementation.
+   */
+  String getConnectorTaskPath(String connector, String task);
```

Comment:
If we are anyways passing connector and task to retrieve instance in any custom method from datastream store, that implies that the path consists of connector and task. This anyways has the same assumption that any other store implementation would need the same arguments to implement that method. 

I just wrote a simple function instead to retrieve the paths, which could be reused (in getters/setters) and so a generic getter (of raw data) could handle retrieving data using these functions.

And since the brooklin-server accesses the datastream store, we can prevent data leaks, as it sits on a layer before the client library accesses it.

Just wanted to share my thought process, let me know @vmaheshw @jzakaryan if you still feel otherwise, I'll update the code :)

Reply:
My recommendation is to just expose one method here `getDatastreamTaskInstance(String connector, String task)`. The input parameters are justified. If you want you can pass Datastream instead of connector.

Since this is in open-source, brooklin-server layer is not sufficient to ensure that there is no data leaks.